src/main/java/uk/gov/ons/census/action/model/repository/CaseRepository.java

```diff
@@ -1,13 +1,13 @@
 package uk.gov.ons.census.action.model.repository;
 
+import java.util.Optional;
 import java.util.UUID;
-import java.util.stream.Stream;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
 import org.springframework.data.rest.core.annotation.RestResource;
 import uk.gov.ons.census.action.model.entity.Case;
 
 @RestResource(exported = false)
 public interface CaseRepository extends JpaRepository<Case, UUID>, JpaSpecificationExecutor<Case> {
-  Stream<Case> findByActionPlanId(String actionPlanId);
+  Optional<Case> findByCaseId(UUID caseId);
```

Comment:
You will need to add an index on the `caseId` column in the cases table otherwise this will table scan

Reply:
Added